src/MixerBoard.cpp

```diff
@@ -601,63 +627,53 @@ void MixerTrackCluster::UpdateMeter(const double t0, const double t1)
    // of the meter display, but instead consult PlaybackPolicy
 
    const auto pTrack = GetWave();
-   auto startSample = (sampleCount)((pTrack->GetRate() * t0) + 0.5);
-   auto scnFrames = (sampleCount)((pTrack->GetRate() * (t1 - t0)) + 0.5);
+
+   // Don't throw on read error in this drawing update routine
+   constexpr auto mayThrow = false;
+   mSampleView = pTrack->GetSampleView(t0, t1, mayThrow);
 
    // Expect that the difference of t1 and t0 is the part of a track played
    // in about 1/20 second (ticks of TrackPanel timer), so this won't overflow
-   auto nFrames = scnFrames.as_size_t();
+   const auto nFrames = GetNumSamplesInView(mSampleView[0]);
 
    Floats tempFloatsArray{ nFrames };
```

Comment:
The deleted `as_size_t` is now instead inside the function called above.

I'm not convinced there won't even be an extreme stretch ratio.

If the assertion ever fired in a debug build -- the release build might continue, with a wrapped, small value.  No harm except an inaccurate meter that put too few floats in its update queue.

But with a less extreme value, which doesn't cause the assertion, still a large size_t value might not be allocable and the release build would crash.

So how about this little bit of bullet-proofing here.

```
Floats tempFloatsArray;
try {
   tempFloatsArray.reinit(nFrames);
}
catch(const std::bad_alloc &) {
   return;
}
```

Reply:
Do this also when constructing `meterFloatsArray`.

Reply:
I just tried to crash it, calling `Floats tempFloatsArray{ std::numeric_limits<size_t>::max() };`. The exception was caught by the `Publisher` code, which visits its observers within a try/catch block. This is the case here, since this scope reacts to an `OnTimer` change. So 🤷

Reply:
But what use would this be, since the exception is already caught above ? I'm not sure we're understanding each other.